### ISSUE TITLE
Remove Dependency on jQuery

### ISSUE THREAD

**Issue by davidbarratt** [wikimedia/jquery.i18n#153]
Is there any reason this library needs jQuery?

Perhaps the majority of this library should be moved to a generic JavaScript library and this package can wrap the new library with jQuery (in order to preserve backwards compatibility).

**Comment by santhoshtr** [comment]
Yes, Agree. See https://github.com/wikimedia/jquery.i18n/issues/83. I started some initial work on this and now stalled due to time constraints.

**Comment by davidbarratt** [comment]
I ended up creating React bindings for this library:
https://github.com/wikimedia/InteractionTimeline/tree/master/client/src/components/i18n
It can (and probably should) be split out into a separate library.

So in the end, I think it would be best if we had a JavaScript i18n library, and then a library that provides jQuery bindings and another that provides React bindings (or any other library someone wants to write bindings for)

**Comment by davidbarratt** [comment]
I've moved the React bindings into a separate library: https://github.com/wikimedia/react.i18n

**Comment by santhoshtr** [comment]
I am working on this again - to build a generic, modern js library https://github.com/santhoshtr/banana-i18n 

Please take a look.

**Comment by davidbarratt** [comment]
AMAZING!

I created a few issues for it. I don't see anything major. When it's published on npm I'll put it in https://github.com/wikimedia/react.i18n and see if the tests pass (I don't see why they wouldn't).

Thank you @santhoshtr for doing this!
